lib/lhm/column_with_sql.rb

```diff
@@ -18,9 +18,10 @@ def self.column_factory
     #
     # @param name [String, Symbol]
     # @param definition [String]
-    def initialize(name, definition)
+    def initialize(name, definition, connection)
       @name = name
       @definition = definition
+      @connection = connection
```

Comment:
Would you be ok that instead of injecting it constructor I will just use `ActiveRecord::Base.connection.lookup_cast_type` in `ColumnWithSql#column` method? I don't think we can avoid getting this cast_type from connection...